graspologic/embed/svd.py

```diff
@@ -285,7 +298,12 @@ def selectSVD(X, n_components=None, n_elbows=2, algorithm="randomized", n_iter=5
         V = V[idx, :]
 
     elif algorithm == "randomized":
-        U, D, V = sklearn.utils.extmath.randomized_svd(X, n_components, n_iter=n_iter)
+        # for some reason, randomized_svd defaults random_state to 0 if not provided
+        # which is weird because None is a valid starting point too
```

Comment:
so are you trying to stick with their convention?

Reply:
I think it actually broke some unit tests when I changed it to None so I did this song and dance around it.  I'll change it to None and see what we end up with - it may be that we go with it in the short term and fix it in another PR?

Reply:
Yeah, it's breaking more than a few.  I think we can and should fix this, but it wasn't something I was trying to do in this PR - it already touched on too many classes, I think!

Results of commenting out line 303:

```
============================================================================= FAILURES ==============================================================================
_________________________________________________________________________ TestGMP.test_sim __________________________________________________________________________

self = <tests.test_match.TestGMP object at 0x1448ff640>

    def test_sim(self):
        n = 150
        rho = 0.9
        n_per_block = int(n / 3)
        n_blocks = 3
        block_members = np.array(n_blocks * [n_per_block])
        block_probs = np.array(
            [[0.2, 0.01, 0.01], [0.01, 0.1, 0.01], [0.01, 0.01, 0.2]]
        )
        directed = False
        loops = False
        A1, A2 = sbm_corr(
            block_members, block_probs, rho, directed=directed, loops=loops
        )
        ase = AdjacencySpectralEmbed(n_components=3, algorithm="truncated")
        x1 = ase.fit_transform(A1)
        x2 = ase.fit_transform(A2)
        xh1 = SignFlips().fit_transform(x1, x2)
        S = xh1 @ x2.T
        res = self.barygm.fit(A1, A2, S=S)

>       assert 0.7 <= (sum(res.perm_inds_ == np.arange(n)) / n)
E       assert 0.7 <= (78 / 150)
E        +  where 78 = sum(array([ 86,  ...47, 148, 149]) == array([  0,  ...47, 148, 149])
E           Use -v to get the full diff)

tests/test_match.py:251: AssertionError
___________________________________________________________________ TestDCSBM.test_DCSBM_nparams ____________________________________________________________________

self = <tests.test_models.TestDCSBM object at 0x131e2c6d0>

    def test_DCSBM_nparams(self):
        n_verts = 3000
        n_class = 4
        graph = self.g
        labels = self.labels
        e = DCSBMEstimator(directed=True)
        e.fit(graph)
>       assert e._n_parameters() == (n_verts + n_class - 1 + n_class ** 2)
E       assert 3071 == (((3000 + 4) - 1) + (4 ** 2))
E        +  where 3071 = <bound method DCSBMEstimator._n_parameters of DCSBMEstimator()>()
E        +    where <bound method DCSBMEstimator._n_parameters of DCSBMEstimator()> = DCSBMEstimator()._n_parameters

tests/test_models.py:422: AssertionError
______________________________________________________________________ TestOmni.test_diag_aug _______________________________________________________________________

self = <tests.test_omni.TestOmni testMethod=test_diag_aug>

    def test_diag_aug(self):
        np.random.seed(5)
        n = 100
        p = 0.25

        graphs_list = [er_np(n, p) for _ in range(2)]
        graphs_arr = np.array(graphs_list)

        # Test that array and list inputs results in same embeddings
        omni_arr = OmnibusEmbed(diag_aug=True).fit_transform(graphs_arr)
        omni_list = OmnibusEmbed(diag_aug=True).fit_transform(graphs_list)

>       np.testing.assert_array_equal(omni_list, omni_arr)
E       AssertionError:
E       Arrays are not equal
E
E       Mismatched elements: 1200 / 1200 (100%)
E       Max absolute difference: 1.415089
E       Max relative difference: 212.39581875
E        x: array([[[ 0.508661, -0.044478, -0.16163 ,  0.092923,  0.106579,
E                -0.216692],
E               [ 0.436264,  0.371707,  0.193453, -0.028346, -0.123348,...
E        y: array([[[ 0.508661, -0.015476, -0.238229,  0.027816,  0.019887,
E                -0.283967],
E               [ 0.436264,  0.338461,  0.145553, -0.009014, -0.131633,...

tests/test_omni.py:133: AssertionError
__________________________________________________ TestAdjacencySpectralEmbed.test_transform_closeto_fit_transform __________________________________________________

self = <tests.test_spectral_embed.TestAdjacencySpectralEmbed testMethod=test_transform_closeto_fit_transform>

    def test_transform_closeto_fit_transform(self):
        atol = 0.15
        for diag_aug in [True, False]:
            for g, A in self.testgraphs.items():
                ase = AdjacencySpectralEmbed(
                    n_components=2, diag_aug=diag_aug, svd_seed=9001
                )
                ase.fit(A)
                Y = ase.fit_transform(A)
                if isinstance(Y, np.ndarray):
                    X = ase.transform(A)
                    np.testing.assert_allclose(X, Y, atol=atol)
                elif isinstance(Y, tuple):
                    with self.assertRaises(TypeError):
                        X = ase.transform(A)
                    X = ase.transform((A.T, A))
                    np.testing.assert_allclose(X[0], Y[0], atol=atol)
>                   np.testing.assert_allclose(X[1], Y[1], atol=atol)
E                   AssertionError:
E                   Not equal to tolerance rtol=1e-07, atol=0.15
E
E                   Mismatched elements: 1 / 40 (2.5%)
E                   Max absolute difference: 0.18669627
E                   Max relative difference: 5.41864066
E                    x: array([[ 1.523434, -0.9013  ],
E                          [ 1.624558, -0.937225],
E                          [ 0.753984, -0.933442],...
E                    y: array([[ 1.586404, -0.957384],
E                          [ 1.711405, -0.99416 ],
E                          [ 0.791389, -0.990951],...

tests/test_spectral_embed.py:178: AssertionError
=======================================================================
```

Reply:
so looks like many of these were just fragile to exactly what SVD they were getting out with the implicitly set randomized seed. just off the top of my head - 
 - TestMatch: it's possible this test was super fragile to the SVD seed, which we should fix?
 - DCSBM: no idea, maybe was also fragile to the SVD seed which messed up the embedding which led to wrong # of parameters?
 - Omni: can't expect exact same SVDs if using randomized and the seeds are not set, should probably just set the seed for this test.
 - ASE transform: probably should just set the seed here also for the same reason as the above.

Reply:
Something else is still weird in the ase transform - it *is* setting the seed, yet it isn't getting the same results.  we don't create a random state of our own with the seed - we use the seed as an int every time, so any time we call select_svd with randomized algorithm, we're going to use the seed value 9001.  so... this shouldn't be failing, and I'm not entirely sure what we're trying to test with it.  

This just underscores that I'm going to need someone else to look into these with more detail.  I can probably fix the omni one - I'm pretty sure I just missed that one.  The others just left me with question marks firing off like crazy.

Reply:
#815 btw